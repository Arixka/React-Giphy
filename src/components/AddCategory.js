import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    console.log('hangleSubmit', inputValue)
    e.preventDefault();
    if(inputValue.trim().length>2){
      // console.log(setCategories(cat=>[inputValue,...cat]))
      setInputValue('')
    }

  }
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type='text'
        placeholder='Add Category'
        value={inputValue}
        onChange={handleInputValue}
      ></input>
    </form>
  )
}

AddCategory.propTypes= {
  setCategories: PropTypes.func.isRequired
}
