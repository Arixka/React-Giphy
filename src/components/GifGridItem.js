import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title,url}) => {
 // console.log(id, title,url)
  return (
    <div className='card animate__animated animate__fadeIn animate__delay-2s'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}