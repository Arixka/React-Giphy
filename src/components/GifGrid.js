import React from 'react'
import PropTypes from 'prop-types'
import {useFetchGifs} from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifs(category)

  return (
    <>
      {loading && <p className="animate__animated animate__flash animate__delay-2s">Loading...</p>}

      <h3 className="animate__animated animate__fadeIn animate__delay-2s">
        {category}
      </h3>
      <div className='card-grid'>
        {images.map((img) =>
          <GifGridItem
            key={img.id}
            {...img} />//...img mandamos cada imagen con sus propiedades 
        )}

      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}