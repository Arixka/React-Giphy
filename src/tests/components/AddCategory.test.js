import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('pruebas con AddCategory', () => {

  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })


  test('debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe cambiar la caja de texto', () => {
    const input = wrapper.find('input')
    const value = 'Hola mundo'
    input.simulate('change', { target: { value } })

    expect(wrapper.find('p').text().trim()).toEqual(value)
  })
  test('No debe postear la informacion con submit, pq no llamamos a setcatefories', () => {

    wrapper.find('form').simulate('submit', { preventDefault() { } })
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('Debe de llamar a setCategories y limpiar la caja de texto',()=>{

    const value = 'contenido del input'
    wrapper.find('input').simulate('change',{ target: { value }})//simulamos el input
    wrapper.find('form').simulate('submit', { preventDefault() { }})//simulamos el submit
   // expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})



//value del input debe ser vacio ''