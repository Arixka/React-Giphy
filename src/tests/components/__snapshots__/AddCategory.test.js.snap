// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con AddCategory debe renderizar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
